src/overlays/actors/ovl_En_Dnp/z_en_dnp.c

```diff
@@ -22,61 +65,454 @@ const ActorInit En_Dnp_InitVars = {
     (ActorFunc)EnDnp_Draw,
 };
 
-
-// static ColliderCylinderInit sCylinderInit = {
-static ColliderCylinderInit D_80B3DC80 = {
-    { COLTYPE_HIT1, AT_NONE, AC_NONE, OC1_ON | OC1_TYPE_ALL, OC2_TYPE_1, COLSHAPE_CYLINDER, },
-    { ELEMTYPE_UNK1, { 0x00000000, 0x00, 0x00 }, { 0x00000000, 0x00, 0x00 }, TOUCH_NONE | TOUCH_SFX_NORMAL, BUMP_NONE, OCELEM_ON, },
+static ColliderCylinderInit sCylinderInit = {
+    {
+        COLTYPE_HIT1,
+        AT_NONE,
+        AC_NONE,
+        OC1_ON | OC1_TYPE_ALL,
+        OC2_TYPE_1,
+        COLSHAPE_CYLINDER,
+    },
+    {
+        ELEMTYPE_UNK1,
+        { 0x00000000, 0x00, 0x00 },
+        { 0x00000000, 0x00, 0x00 },
+        TOUCH_NONE | TOUCH_SFX_NORMAL,
+        BUMP_NONE,
+        OCELEM_ON,
+    },
     { 14, 38, 0, { 0, 0, 0 } },
 };
 
+static CollisionCheckInfoInit2 sColChkInfoInit = { 0, 0, 0, 0, MASS_IMMOVABLE };
+
+s32 func_80B3CA20(EnDnp* this) {
+    if ((this->unk_340 == 2) || (this->unk_340 == 9)) {
+        if (func_801378B8(&this->skelAnime, 1.0f) || func_801378B8(&this->skelAnime, 5.0f) ||
+            func_801378B8(&this->skelAnime, 9.0f) || func_801378B8(&this->skelAnime, 13.0f)) {
+            Audio_PlayActorSound2(&this->actor, NA_SE_EN_DEKUHIME_TURN);
+        }
+    } else if ((this->unk_340 == 0x18) || (this->unk_340 == 7)) {
+        if (func_801378B8(&this->skelAnime, 1.0f)) {
+            Audio_PlayActorSound2(&this->actor, NA_SE_EN_DEKUHIME_TURN);
+        }
+    } else if (this->unk_340 == 14) {
+        if (func_801378B8(&this->skelAnime, 7.0f)) {
+            Audio_PlayActorSound2(&this->actor, NA_SE_EN_DEKUHIME_GREET);
+        }
+        if (func_801378B8(&this->skelAnime, 22.0f)) {
+            Audio_PlayActorSound2(&this->actor, NA_SE_EN_DEKUHIME_GREET2);
+        }
+    } else if (this->unk_340 == 8) {
+        if (func_801378B8(&this->skelAnime, 9.0f)) {
+            Audio_PlayActorSound2(&this->actor, NA_SE_EN_DEKUHIME_GREET);
+        }
+        if (func_801378B8(&this->skelAnime, 18.0f)) {
+            Audio_PlayActorSound2(&this->actor, NA_SE_EN_DEKUHIME_GREET2);
+        }
+    } else if ((this->unk_340 == 17) && (this->unk_340 == 18)) {
+        if (func_801378B8(&this->skelAnime, 7.0f) || func_801378B8(&this->skelAnime, 15.0f)) {
+            Audio_PlayActorSound2(&this->actor, NA_SE_EN_DEKUHIME_WALK);
+        }
+    } else if (this->unk_340 == 21) {
+        if (func_801378B8(&this->skelAnime, 17.0f)) {
+            Audio_PlayActorSound2(&this->actor, NA_SE_EN_DEKUHIME_WALK);
+        }
+    } else if (this->unk_340 == 23) {
+        if (func_801378B8(&this->skelAnime, 3.0f)) {
+            Audio_PlayActorSound2(&this->actor, NA_SE_EN_DEKUHIME_WALK);
+        }
+    }
+
+    return 0;
+}
+
+s32 func_80B3CC38(EnDnp* this, s32 arg1) {
+    static ActorAnimationEntryS D_80B3DCB8[] = {
```

Comment:
It's up for debate, but I really don't think it makes sense to have all these animation structs in the function like this. We don't really have a standard for where to put data at this time, but my opinion is that data should be separate from logic/functions unless required for matching (or if it's some small static data like a color or something). Interested in input from the Discord on this one.

Reply:
Zel said to move data into functions wherever possible in other PR, so I've been trying to do it.